app/labor/cache_buster.rb

```diff
@@ -1,9 +1,12 @@
-class CacheBuster
+module CacheBuster
```

Comment:
I'm going to throw another idea in the mix here.

I believe that this CacheBuster should be a class and have state(though it currently does not). Right now it's just a collection of a bunch of different cache-busting methods. I think another refactor option here would be to make CacheBuster a class that busts a "BustableObject" (horrible name!). Then that BustableObject holds the logic it needs to know how to bust its own cache and the CacheBuster is simply a class for calling those methods to bust it. That's how I saw this working in my head so I figured I would throw it out there.

Reply:
> Right now it's just a collection of a bunch of different cache-busting methods. 

Which I think is fine. Sometimes a collection of functions is all one needs, like in a Python module that doesn't define any classes as @rhymes mentioned above. 

If I understand you correctly — please correct me if I'm wrong — you're suggesting something like

```ruby
class CacheBuster
   def initialize(klass)
      @bustable = klass
   end

   def bust
    @bustable.bust
  end
end
```

This would be called like `CacheBuster.new(PostCacheBuster).bust`  which somehow seems more complicated than just calling `PostCacheBuster.bust` directly. 

Or maybe you were suggesting something more like this?

```ruby
class CacheBuster
  def bust_post_cache
    PostCachebuster.bust
  end
end
```

But then this would require adding a new wrapper method to `CacheBuster` for every specific busting class that is added and I'm not sure what benefit is derived from this extra effort?

As I said, please correct me if I misunderstood your comment, but I don't think that a bunch of related functions thrown into a module is somehow worse than a class + cache, especially if there's no state that needs to persist between method calls of the same instance before it goes out of scope.

Reply:
Your first thought is exactly what I mean and the advantage of having the cache busting logic within the bustable objects is that you can use that for more than just cache-busting. You could ideally use that information to help you set the cache, expire the cache, bust the cache manually etc. 

I think with this change, we could still do something like that in the future on top of this so I'm good with it!

Reply:
Glad I got what you were aiming for. 

> the advantage of having the cache busting logic within the bustable objects is that you can use that for more than just cache-busting

I got that part, it's more that I don't quite see the purpose of `CacheBuster` itself in such a scenario. If one already needs to know the name of the "bustable" to inject it, why not just invoke the necessary method directly? Of course one could hack together something like  the following `method_missing`, but that's exactly the sort of metaprogramming that gives Rails a bad name.

```ruby
class CacheBuster
  def method_missing(m, *args, &blk)
    match = m.match(/^bust_(\w+)/)
    return super unless match

    buster = "#{match[1].classify}Buster"
    buster.bust
  end
end
```